web/src/main/java/org/springframework/security/web/savedrequest/CookieRequestCache.java

```diff
@@ -73,7 +73,13 @@ public SavedRequest getRequest(HttpServletRequest request, HttpServletResponse r
 		if (savedRequestCookie == null) {
 			return null;
 		}
-		String originalURI = decodeCookie(savedRequestCookie.getValue());
+		String originalURI;
+		try {
+			originalURI = decodeCookie(savedRequestCookie.getValue());
+		}
+		catch (IllegalArgumentException ex) {
+			return null;
```

Comment:
Let's please at least log the exception so that it isn't lost. `DEBUG` should be suitable.